Test/XLTest.cs

```diff
@@ -273,6 +273,46 @@ public static void XlTest_DiffCrosslinkSites()
             Assert.AreEqual(newPsms.Count(), 1);
         }
 
+        /// <summary>
+        /// Verifies that crosslinker is generated properly
+        /// </summary>
+        [Test]
+        public static void CrosslinkCreateTest()
+        {
+            Assert.That((XLSearchTask.GenerateUserDefinedCrosslinker(new XlSearchParameters())).GetType().Equals(typeof(CrosslinkerTypeClass)));
+        }
+
+        /// <summary>
+        /// Makes sure helper methods that generate indices function properly
+        /// </summary>
+        [Test]
+        public static void XLSearchWithGeneratedIndices()
+        {
+            XLSearchTask xlSearchTask = new XLSearchTask();
+            string myFile = Path.Combine(TestContext.CurrentContext.TestDirectory, @"XlTestData\BSA_DSS_23747.mzML");
+            string myDatabase = Path.Combine(TestContext.CurrentContext.TestDirectory, @"XlTestData\BSA.fasta");
+            string folderPath = Path.Combine(TestContext.CurrentContext.TestDirectory, @"TestXLSearch");
+            DbForTask db = new DbForTask(myDatabase, false);
+
+            //creates .params files if they do not exist
+            xlSearchTask.RunTask(folderPath, new List<DbForTask> { db }, new List<string> { myFile }, "normal");
+            //tests .params files
+            xlSearchTask.RunTask(folderPath, new List<DbForTask> { db }, new List<string> { myFile }, "normal");
```

Comment:
can you write the results of these two tasks to two different directories and then check that the output is equal?